fail2ban/server/banmanager.py

```diff
@@ -264,11 +262,25 @@ def createBanTicket(ticket):
 	def addBanTicket(self, ticket):
 		try:
 			self.__lock.acquire()
-			if not self._inBanList(ticket):
-				self.__banList.append(ticket)
-				self.__banTotal += 1
-				return True
-			return False
+			# check already banned
+			for i in self.__banList:
```

Comment:
and `goto` is prohibited... guys, you are excellent trolls :)
`i` is meanwhile anything in the little cycle, particularly in the language with dynamic type system.